### PR TITLE
refactor(config): Move configs to components

### PR DESCRIPTION
**Pull request by oxarbitrage**
## Motivation

This was inspired by https://github.com/ZcashFoundation/zebra/issues/5436 where there is a broken link and we don't know why. 

I am not sure if this will fix the link issue but in my opinion the docs about the config and also the code will be a bit more clear if each `Config` is where it needs to be.

config

## Solution

- Move the config code to each component (https://github.com/ZcashFoundation/zebra/commit/c21d0be05848ce1437f0c93a2a50506c135d3b04) and make needed fixes to make them work that way.
- Remove trailing slash from zebrad Cargo.toml when importing consensus (https://github.com/ZcashFoundation/zebra/commit/68a681ee904ee5db26b1f2cff9f7f09dc2ee92d2)
- Remove a doc warning that will be visible when running `cargo doc`. This is because the code the link is pointing to is behind a rust feature, so i prefer to just remove the link in those cases (https://github.com/ZcashFoundation/zebra/commit/5572f748efea36091438217984b2e1d4d73e2b1e)

## Review

I think anyone can review. In regards to the priority i think it can depend, we could want to add it to the new audit tag we are going to do. If so the priority can be medium, low otherwise.

### Reviewer Checklist

  - [ ] Will the PR name make sense to users?
    - [ ] Does it need extra CHANGELOG info? (new features, breaking changes, large changes)
  - [ ] Are the PR labels correct?
  - [ ] Does the code do what the ticket and PR says?
  - [ ] How do you know it works? Does it have tests?

### CODE REVIEW

File: zebrad/src/components/sync.rs [hunk 1]
```diff
@@ -204,6 +205,83 @@ const SYNC_RESTART_DELAY: Duration = Duration::from_secs(67);
 /// a denial of service on those peers.
 const GENESIS_TIMEOUT_RETRY: Duration = Duration::from_secs(5);
 
+/// Sync configuration section.
```
Comment (teor2345) [inline zebrad/src/components/sync.rs:208]: Optional: This module is already over 1000 lines long, would you mind putting the config in a `sync/config.rs` submodule?

**Review by teor2345 — COMMENTED** [review]
Looks good, just wondering if we can make a submodule to keep a file from getting too big?

What's the priority of this PR?

**Comment by teor2345** [comment]
> I am not sure if this will fix the link issue but in my opinion the docs about the config and also the code will be a bit more clear if each `Config` is where it needs to be.

Can we build the docs and check if it fixes the links?

**Comment by oxarbitrage** [comment]
> Can we build the docs and check if it fixes the links?

I don't think we should spend time doing this right now. Links work locally and making the CI build docs for this PR, push them to the website (https://doc.zebra.zfnd.org) could be hard (i honestly don't know).

The idea here is to make the refactor because i think it is the right thing to do, then when the CI build the docs we can figure out if the link was fixed and close the related issue or keep it open, but we will have the refactor done (which i think we should do anyway).

**Comment by teor2345** [comment]
> > Can we build the docs and check if it fixes the links?
> 
> I don't think we should spend time doing this right now. Links work locally and making the CI build docs for this PR, push them to the website (https://doc.zebra.zfnd.org) could be hard (i honestly don't know).
> 
> The idea here is to make the refactor because i think it is the right thing to do, then when the CI build the docs we can figure out if the link was fixed and close the related issue or keep it open, but we will have the refactor done (which i think we should do anyway).

Sure, that makes sense, let's give it a shot!

**Review by teor2345 — APPROVED** [review]

**Pull request merged**